app/scripts/modules/core/src/presentation/spel/SpelToggle.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+
+import { Tooltip } from 'core/presentation';
+
+interface ISpelToggleProps {
+  inputMode: SpelAwareInputMode;
+  onClick: () => void;
+}
+
+export enum SpelAwareInputMode {
+  DEFAULT = 'DEFAULT',
+  FREEFORM = 'FREEFORM',
+}
+
+export function SpelToggle(props: ISpelToggleProps) {
+  const inputModeLabels: { [k in SpelAwareInputMode]: string } = {
+    [SpelAwareInputMode.DEFAULT]: 'Enter SpEL in freeform input',
+    [SpelAwareInputMode.FREEFORM]: 'Return to default input',
+  };
+
+  return (
+    <Tooltip value={inputModeLabels[props.inputMode]}>
+      <button className="btn btn-sm btn-default" onClick={props.onClick}>
```

Comment:
can we get an active state class on this button?  Greg used yellow in the mocks (which I don't personally love) but he's out until next monday.

Reply:
Great call! Conditionally added the `active` class so it would get the default gray-ish `btn-default:active` styling, but please feel free to change when Greg returns if we feel strongly about the yellow!